### ISSUE TITLE
Chrome crash on scale range [1, 1]

### ISSUE THREAD

**Issue by mario-jerkovic** [leeoniya/uPlot#620]
**uPlot version**: 1.6.17
**Browser**: Chrome 95.0.4638.69
**OS**: macOS

**Bug description**:
If you set scale range to equal numbers like [1,1], [2,2] and etc.. except [0,0] memory usage skyrockets and tab crashes.
![Screenshot 2021-11-16 at 14 08 49](https://user-images.githubusercontent.com/16222284/141990947-f3b7d27f-f666-4ab9-a3d7-84dd1215c5d4.png)

**Expected behaviour**:
In previous versions of uPlot if scale has equal range values it wouldn't crash the tab and scale label would be hidden

**To reproduce**:
In any of the project examples set scale range to [1,1]

**Comment by Andro172** [comment]
I have the same issue, please fix this 🙏

**Comment by leeoniya** [comment]
will take a look today.

1. out of curiosity, can this be triggered outside of manually returning a min == max range?
2. is there a logical expectation that this is a valid range to return, besides that it used to work? like, in what scenario woild you need to do this?

**Comment by mario-jerkovic** [comment]
@leeoniya 

This will happen when Chart receives one X value and one Y value and scale range function by default will return min: 1 and max: 1, example below:

```
data: [
    [
        1_636_668_028_800,
    ],
    [
        0, // uPlot range function returns min: 0, max: 0 which is fine
    ],
    [
        1, // uPlot range function returns min: 1, max: 1 which crashes 
    ],
]
```

**Comment by leeoniya** [comment]
right, which is why my first question was whether this happened without you manually returning a nonsense scale. you said "Yes", but your demo is "No". in this case, you are manually returning a nonsense scale range; you can also probably also cause a crash by returning `["foo", -5]`.

`uPlot.rangeNum()` has these guards built in, and internally that is what it uses _unless_ you supply your own `scale.range` manually. in this case it is up to you to return a range that is non-0, numeric, and has max > min.

for now, you should guard against either manually, or by using `uPlot.rangeNum()`. e.g. https://github.com/leeoniya/uPlot/blob/1692dc3556a9b8c883abf8488859f4272ceb6ee4/demos/no-data.html#L32-L37

i will consider adding a guard for this specific case to avoid an infinite loop (cause these are hard to debug), but there are a lot of other places in the API when you can do the wrong thing and cause all sorts of unexpected behavior.

**Comment by leeoniya** [comment]
soon there's going to be some significant internal rework to how scale auto-ranging works to take into account axis tick density requirements and auto-range in a way that ensures there is a tick above the max and below the min (right now this is not guaranteed and can lead to less than ideal situations [1]). some initial work is in [2]

[1] https://github.com/grafana/grafana/discussions/39307
[2] https://github.com/leeoniya/uPlot/tree/ticks-axis-incrs

**Comment by leeoniya** [comment]
flat data has not crashed for a long time now

"soon" was was apparently, 5 years in leon time 😅 

scale ranging got this major internal refactor in the  last couple weeks on `master`, will be in the upcoming 1.7.0
